README.md

```diff
@@ -20,3 +20,19 @@ randomly without memory instead of systematically trying them?
 
 Describe your reasoning and the conclusion you've come to. Your reasoning is the
 most important part. Add your answer to this markdown file.
+
+This algorithm has a complexity of $O(n!*n)$ this is because there are n! possible permutations of any given array, and checksort is called for every permutation, and that is called for the length of the list n. 
+
+The best case input is an already sorted list, there will be one permutation to check, and it will immediately determine correctness.
+The worst case input would be a function that is in the last checked combination, this varies based on the length of the input array, but for an input array of [a,b,c,d], the worst possible input would be [b,c,d,a] as that would be the last permutation the algorithm would check.
+
+The complexity does not change with random permutations. Checksort is still called for the length of the array, and there are still always n! possiblities, meaning our worst case is still $n*n!$
```

Comment:
Is there any guarantee that random would ever find the correct permutation?

Reply:
There's no guarantee that it would find the correct permutation, so would that make our worst case infinity?

Reply:
Sounds like it's going in the right direction.

Reply:
Gave further reasoning in the readme